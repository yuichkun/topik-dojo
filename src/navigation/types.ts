/**
 * ナビゲーション型定義
 */

import type { NativeStackScreenProps } from '@react-navigation/native-stack';

// ルートスタックのパラメータ定義
export type RootStackParamList = {
  Top: undefined;
  LearningModeSelection: {
    level: number;
  };
  UnitSelection: { 
    level: number; 
  };
  Learning: { 
    level: number; 
    unitNumber: number; 
  };
  TestModeSelection: { 
    level: number; 
  };
  // TODO: 他の画面のパラメータ定義を追加
  // Results: { level: number };
  // ListeningUnitSelection: { level: number; mode: 'listening' };
  // ReadingUnitSelection: { level: number; mode: 'reading' };
  // ListeningTest: { level: number; unitId: string };
  // ReadingTest: { level: number; unitId: string };
  // Review: undefined;
};

// 各画面のプロップス型定義
export type TopScreenProps = NativeStackScreenProps<RootStackParamList, 'Top'>;
export type LearningModeSelectionScreenProps = NativeStackScreenProps<RootStackParamList, 'LearningModeSelection'>;
export type UnitSelectionScreenProps = NativeStackScreenProps<RootStackParamList, 'UnitSelection'>;
export type LearningScreenProps = NativeStackScreenProps<RootStackParamList, 'Learning'>;
export type TestModeSelectionScreenProps = NativeStackScreenProps<RootStackParamList, 'TestModeSelection'>;

declare global {
  namespace ReactNavigation {
    interface RootParamList extends RootStackParamList {}
  }
}