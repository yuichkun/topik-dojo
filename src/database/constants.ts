// テーブル名の定数定義（型安全性向上）
export enum TableName {
  UNITS = 'units',
  WORDS = 'words',
  SRS_MANAGEMENT = 'srs_management',
  LEARNING_STATUS = 'learning_status',
  TEST_RESULTS = 'test_results',
  TEST_QUESTIONS = 'test_questions',
  REVIEW_HISTORY = 'review_history',
  LEARNING_PROGRESS = 'learning_progress',
}

// データベース設定の定数定義
export const DATABASE_CONFIG = {
  name: 'TopikDojo',
  testName: ':memory:',
} as const;