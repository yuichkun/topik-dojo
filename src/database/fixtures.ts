// テスト用のfixture データ

export const testWords = [
  {
    id: 'word_1',
    korean: '안녕하세요',
    japanese: 'こんにちは',
    example_korean: '안녕하세요. 반갑습니다.',
    example_japanese: 'こんにちは。お会いできて嬉しいです。',
    grade: 1,
    grade_word_number: 1,
  },
  {
    id: 'word_2', 
    korean: '감사합니다',
    japanese: 'ありがとうございます',
    example_korean: '정말 감사합니다.',
    example_japanese: '本当にありがとうございます。',
    grade: 1,
    grade_word_number: 2,
  },
  {
    id: 'word_3',
    korean: '죄송합니다',
    japanese: 'すみません',
    example_korean: '늦어서 죄송합니다.',
    example_japanese: '遅れてすみません。',
    grade: 1,
    grade_word_number: 3,
  },
  {
    id: 'word_4',
    korean: '학생',
    japanese: '学生',
    example_korean: '저는 대학생입니다.',
    example_japanese: '私は大学生です。',
    grade: 2,
    grade_word_number: 1,
  },
  {
    id: 'word_5',
    korean: '선생님',
    japanese: '先生',
    example_korean: '우리 선생님은 친절하십니다.',
    example_japanese: '私たちの先生は親切です。',
    grade: 2,
    grade_word_number: 2,
  }
];

export const testSrsData = [
  {
    id: 'srs_1',
    word_id: 'word_1',
    mastery_level: 0,
    ease_factor: 2.5,
    next_review_date: Date.now(), // 今日復習対象
    interval_days: 1,
    mistake_count: 0,
    last_reviewed: null,
    status: 'learning',
  },
  {
    id: 'srs_2',
    word_id: 'word_2',
    mastery_level: 1,
    ease_factor: 2.5,
    next_review_date: Date.now() + (2 * 24 * 60 * 60 * 1000), // 2日後
    interval_days: 3,
    mistake_count: 0,
    last_reviewed: Date.now() - (1 * 24 * 60 * 60 * 1000), // 1日前
    status: 'learning',
  },
  {
    id: 'srs_3',
    word_id: 'word_3',
    mastery_level: 3,
    ease_factor: 2.5,
    next_review_date: Date.now() + (5 * 24 * 60 * 60 * 1000), // 5日後
    interval_days: 6,
    mistake_count: 1,
    last_reviewed: Date.now() - (1 * 24 * 60 * 60 * 1000), // 1日前
    status: 'graduated',
  }
];

// データベースにテストデータを投入するヘルパー関数
export const seedTestData = async (database: any) => {
  await database.write(async () => {
    // 既存データをクリア
    const existingWords = await database.collections.get('words').query().fetch();
    for (const word of existingWords) {
      await word.markAsDeleted();
    }

    const existingSrs = await database.collections.get('srs_management').query().fetch();
    for (const srs of existingSrs) {
      await srs.markAsDeleted();
    }

    // テストデータを投入
    for (const wordData of testWords) {
      await database.collections.get('words').create((word: any) => {
        word._raw.id = wordData.id;
        word.korean = wordData.korean;
        word.japanese = wordData.japanese;
        word.exampleKorean = wordData.example_korean;
        word.exampleJapanese = wordData.example_japanese;
        word.grade = wordData.grade;
        word.gradeWordNumber = wordData.grade_word_number;
      });
    }

    for (const srsData of testSrsData) {
      await database.collections.get('srs_management').create((srs: any) => {
        srs._raw.id = srsData.id;
        srs.wordId = srsData.word_id;
        srs.masteryLevel = srsData.mastery_level;
        srs.easeFactor = srsData.ease_factor;
        srs.nextReviewDate = srsData.next_review_date;
        srs.intervalDays = srsData.interval_days;
        srs.mistakeCount = srsData.mistake_count;
        srs.lastReviewed = srsData.last_reviewed;
        srs.status = srsData.status;
      });
    }
  });
};