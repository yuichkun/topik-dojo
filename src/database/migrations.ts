import { schemaMigrations, createTable, addColumns } from '@nozbe/watermelondb/Schema/migrations';

export default schemaMigrations({
  migrations: [
    // Version 2: UNITSテーブル追加とWORDSテーブル修正
    {
      toVersion: 2,
      steps: [
        // 1. UNITSテーブルを作成
        createTable({
          name: 'units',
          columns: [
            { name: 'grade', type: 'number' },
            { name: 'unit_number', type: 'number' },
            { name: 'created_at', type: 'number' },
            { name: 'updated_at', type: 'number' },
          ],
        }),
        
        // 2. WORDSテーブルに新しいカラムを追加
        addColumns({
          table: 'words',
          columns: [
            { name: 'unit_id', type: 'string' },
            { name: 'unit_order', type: 'number' },
          ],
        }),
      ],
    },
  ],
});