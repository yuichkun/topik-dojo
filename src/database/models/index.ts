// モデルのエクスポート
export { default as Unit } from './Unit';
export { default as Word } from './Word';
export { default as SrsManagement } from './SrsManagement';
export { default as TestResult } from './TestResult';
export { default as TestQuestion } from './TestQuestion';
export { default as ReviewHistory } from './ReviewHistory';
export { default as LearningProgress } from './LearningProgress';

// モデルクラスの配列（Database初期化時に使用）
import Unit from './Unit';
import Word from './Word';
import SrsManagement from './SrsManagement';
import TestResult from './TestResult';
import TestQuestion from './TestQuestion';
import ReviewHistory from './ReviewHistory';
import LearningProgress from './LearningProgress';

export const modelClasses = [
  Unit,
  Word,
  SrsManagement,
  TestResult,
  TestQuestion,
  ReviewHistory,
  LearningProgress,
];