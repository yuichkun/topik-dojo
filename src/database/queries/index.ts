// ユニット関連のクエリ
export * from './unitQueries';

// 単語関連のクエリ
export * from './wordQueries';