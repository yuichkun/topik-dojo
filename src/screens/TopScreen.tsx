/**
 * トップ画面（級選択）
 * 級選択と復習モードの選択を行う画面
 */

import React from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  StatusBar,
  useColorScheme,
  SafeAreaView,
} from 'react-native';
import { useReviewCount } from '../hooks/useReviewCount';
import { TopScreenProps } from '../navigation/types';
import { SCREEN_NAMES } from '../constants/screens';

const TopScreen: React.FC<TopScreenProps> = ({ navigation }) => {
  const { count: reviewCount } = useReviewCount();
  const isDarkMode = useColorScheme() === 'dark';

  // 復習ボタンのタップハンドラ
  const handleReviewPress = () => {
    if (reviewCount === 0) {
      return; // 復習対象が0件の場合は何もしない
    }
    
    navigation.navigate(SCREEN_NAMES.REVIEW);
  };

  // 級選択ボタンのタップハンドラ
  const handleLevelPress = (level: number) => {
    navigation.navigate(SCREEN_NAMES.LEARNING_MODE_SELECTION, { level });
  };

  const levels = [1, 2, 3, 4, 5, 6];

  return (
    <SafeAreaView className="flex-1 bg-white">
      <StatusBar 
        barStyle={isDarkMode ? 'light-content' : 'dark-content'}
        backgroundColor="#ffffff"
      />
      
      {/* アプリタイトル */}
      <View className="items-center pt-16 pb-10">
        <Text className="text-3xl font-bold text-gray-800 tracking-wider">TOPIK道場</Text>
      </View>

      {/* 復習ボタン */}
      <View className="items-center py-8">
        <TouchableOpacity
          className={`
            border-2 rounded-lg px-10 py-4 bg-white
            ${reviewCount === 0 
              ? 'border-gray-300 bg-gray-50' 
              : 'border-blue-500'
            }
          `}
          onPress={handleReviewPress}
          disabled={reviewCount === 0}
        >
          <Text className={`
            text-lg font-semibold
            ${reviewCount === 0 
              ? 'text-gray-400' 
              : 'text-blue-500'
            }
          `}>
            復習 ({reviewCount}語)
          </Text>
        </TouchableOpacity>
      </View>

      {/* 級選択ボタン */}
      <View className="flex-1 justify-center px-5">
        <View className="items-center">
          {/* 上段: 1級, 2級, 3級 */}
          <View className="flex-row justify-between w-full my-4">
            {levels.slice(0, 3).map((level) => (
              <TouchableOpacity
                key={level}
                className="flex-1 border border-gray-800 rounded py-5 mx-2 items-center bg-white"
                onPress={() => handleLevelPress(level)}
              >
                <Text className="text-base font-medium text-gray-800">{level}級</Text>
              </TouchableOpacity>
            ))}
          </View>
          
          {/* 下段: 4級, 5級, 6級 */}
          <View className="flex-row justify-between w-full my-4">
            {levels.slice(3, 6).map((level) => (
              <TouchableOpacity
                key={level}
                className="flex-1 border border-gray-800 rounded py-5 mx-2 items-center bg-white"
                onPress={() => handleLevelPress(level)}
              >
                <Text className="text-base font-medium text-gray-800">{level}級</Text>
              </TouchableOpacity>
            ))}
          </View>
        </View>
      </View>
    </SafeAreaView>
  );
};

export default TopScreen;