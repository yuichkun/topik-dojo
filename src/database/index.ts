import { Database } from '@nozbe/watermelondb';
import SQLiteAdapter from '@nozbe/watermelondb/adapters/sqlite';

import schema from './schema';
import migrations from './migrations';
import { modelClasses } from './models';
import { DATABASE_CONFIG } from './constants';

// SQLiteアダプターの設定
const adapter = new SQLiteAdapter({
  schema,
  migrations,
  jsi: true, // JSI (JavaScript Interface) を有効化（パフォーマンス向上）
  dbName: DATABASE_CONFIG.name, // データベースファイル名を設定から取得（.dbは自動付与される）
});

// データベースインスタンスの作成
const database = new Database({
  adapter,
  modelClasses,
});

export default database;