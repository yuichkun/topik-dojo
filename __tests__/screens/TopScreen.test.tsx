/**
 * @format
 */

import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { Alert, TouchableOpacity, Text } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import TopScreen from '../../src/screens/TopScreen';
import { RootStackParamList } from '../../src/navigation/types';
import { createTestWords, createDueReviews } from '../helpers/databaseHelpers';
import database from '../../src/database';

// AlertのスパイFunction
jest.spyOn(Alert, 'alert');

const Stack = createNativeStackNavigator<RootStackParamList>();

// テスト用のナビゲーションコンテナーを作成
const createTestNavigationContainer = () => {
  const TestNavigationContainer = () => (
    <NavigationContainer>
      <Stack.Navigator initialRouteName="Top">
        <Stack.Screen 
          name="Top" 
          component={TopScreen}
          options={{ headerShown: false }}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
  return TestNavigationContainer;
};

describe('TopScreen', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders correctly', async () => {
    const TestContainer = createTestNavigationContainer();
    let component: ReactTestRenderer.ReactTestRenderer;
    
    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });
    
    expect(component!).toBeDefined();
  });

  test('displays app title "TOPIK道場"', async () => {
    const TestContainer = createTestNavigationContainer();
    let component: ReactTestRenderer.ReactTestRenderer;
    
    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });
    
    const title = component!.root.findByProps({ children: 'TOPIK道場' });
    expect(title).toBeDefined();
  });

  test('displays review button with database count', async () => {
    // テスト用のデータを作成
    const words = await createTestWords(database, [
      { korean: '안녕하세요', japanese: 'こんにちは' },
      { korean: '감사합니다', japanese: 'ありがとうございます' },
      { korean: '죄송합니다', japanese: 'すみません' }
    ]);
    
    // 復習対象のSRSレコードを作成（3語中2語が復習対象）
    await createDueReviews(database, [words[0].id, words[1].id]);
    
    const TestContainer = createTestNavigationContainer();
    let component: ReactTestRenderer.ReactTestRenderer;
    
    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });
    
    // 少し時間を置いてuseEffectが実行されるのを待つ
    await new Promise(resolve => setTimeout(resolve, 100));
    
    // 復習ボタンのテキストを確認（2語が復習対象）
    const reviewButton = component!.root.findAllByType(TouchableOpacity)[0];
    const reviewButtonText = reviewButton.findByType(Text);
    expect(reviewButtonText.props.children).toEqual(['復習 (', 2, '語)']);
  });

  test('displays all 6 level buttons', async () => {
    const TestContainer = createTestNavigationContainer();
    let component: ReactTestRenderer.ReactTestRenderer;
    
    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });
    
    // 1級から6級までのボタンが表示されることを確認
    const levelButtons = component!.root.findAllByType(TouchableOpacity);
    // 復習ボタンも含まれるので、級ボタンは1番目以降
    const actualLevelButtons = levelButtons.slice(1);
    
    expect(actualLevelButtons).toHaveLength(6);
    
    for (let i = 0; i < 6; i++) {
      const buttonText = actualLevelButtons[i].findByType(Text);
      expect(buttonText.props.children).toEqual([i + 1, '級']);
    }
  });

  test('review button is disabled when no reviews available', async () => {
    const TestContainer = createTestNavigationContainer();
    let component: ReactTestRenderer.ReactTestRenderer;
    
    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });
    
    // 復習ボタンを取得
    const reviewButton = component!.root.findAllByType(TouchableOpacity)[0];
    
    // ボタンが無効化されていることを確認（0語の状態）
    const reviewButtonText = reviewButton.findByType(Text);
    expect(reviewButtonText.props.children).toEqual(['復習 (', 0, '語)']);
    
    // タップしてもアラートが呼ばれないことを確認
    await ReactTestRenderer.act(async () => {
      reviewButton.props.onPress();
    });
    
    expect(Alert.alert).not.toHaveBeenCalled();
  });

  test('level button can be pressed', async () => {
    const TestContainer = createTestNavigationContainer();
    let component: ReactTestRenderer.ReactTestRenderer;
    
    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });
    
    // 1級ボタンを取得してタップ（復習ボタンの次のTouchableOpacity）
    const levelButtons = component!.root.findAllByType(TouchableOpacity);
    const firstLevelButton = levelButtons[1]; // インデックス0は復習ボタン
    
    // ボタンのonPressプロパティが定義されていることを確認
    expect(firstLevelButton.props.onPress).toBeDefined();
    
    // ボタンが押せることを確認（エラーが発生しないことを確認）
    await ReactTestRenderer.act(async () => {
      firstLevelButton.props.onPress();
    });
    
    // ナビゲーションが正しく動作することを想定（現在は実際のナビゲーションはテストしない）
    expect(true).toBe(true);
  });

  test('matches snapshot', async () => {
    const TestContainer = createTestNavigationContainer();
    let component: ReactTestRenderer.ReactTestRenderer;
    
    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });
    
    expect(component!.toJSON()).toMatchSnapshot();
  });
});