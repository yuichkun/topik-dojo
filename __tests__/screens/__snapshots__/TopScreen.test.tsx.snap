// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopScreen matches snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      fullScreenSwipeShadowEnabled={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      hasLargeHeader={false}
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillAppear={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "zIndex": undefined,
          },
        ]
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <RCTSafeAreaView
            className="flex-1 bg-white"
          >
            <View
              className="items-center pt-16 pb-10"
            >
              <Text
                className="text-3xl font-bold text-gray-800 tracking-wider"
              >
                TOPIK道場
              </Text>
            </View>
            <View
              className="items-center py-8"
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  className="
            text-lg font-semibold
            text-gray-400
          "
                >
                  復習 (
                  0
                  語)
                </Text>
              </View>
            </View>
            <View
              className="flex-1 justify-center px-5"
            >
              <View
                className="items-center"
              >
                <View
                  className="flex-row justify-between w-full my-4"
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text
                      className="text-base font-medium text-gray-800"
                    >
                      1
                      級
                    </Text>
                  </View>
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text
                      className="text-base font-medium text-gray-800"
                    >
                      2
                      級
                    </Text>
                  </View>
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text
                      className="text-base font-medium text-gray-800"
                    >
                      3
                      級
                    </Text>
                  </View>
                </View>
                <View
                  className="flex-row justify-between w-full my-4"
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text
                      className="text-base font-medium text-gray-800"
                    >
                      4
                      級
                    </Text>
                  </View>
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text
                      className="text-base font-medium text-gray-800"
                    >
                      5
                      級
                    </Text>
                  </View>
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text
                      className="text-base font-medium text-gray-800"
                    >
                      6
                      級
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </RCTSafeAreaView>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Top"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
